### PR TITLE
Add .github/workflows/build.yml

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
Just updated the CI setup to use the latest (Dec 13) official swift.org "main" nightly toolchain, which has the two C++ interop fixes that were blocking the build (https://github.com/swiftlang/swift/pull/78100, https://github.com/swiftlang/swift/pull/78147). The CI is green, the build passes now 🥳 

Resolves a bunch reported issues:

**Review by rauhul — APPROVED** [review]
Please give this a nice commit message when merging!

File: led-blink/main/BridgingHeader.h [hunk 1]
```diff
@@ -38,6 +38,8 @@
 //
 // connectedhomeip/src/credentials/FabricTable.h:82:69: error: use of undeclared identifier 'strnlen'
 extern "C" size_t strnlen(const char *s, size_t maxlen);
+// esp-matter/components/esp_matter/esp_matter_client.h:57:26: error: use of undeclared identifier 'strdup'
+extern "C" char *strdup(const char *s1);
```
Comment (rauhul) [inline led-blink/main/BridgingHeader.h:42]: Is there a reason to do this instead of #include <string.h>

Comment (kubamracek) [reply led-blink/main/BridgingHeader.h:42]: I didn't spend time investigating why, but `#include <string.h>` didn't resolve the problem.

**Pull request merged**